### DISCUSSION TITLE
Update dependencies with command line?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [PhilDL/remix-gospel-stack#29] [Q&A]
I was wondering if there is some way to update npm packages across multiple /packages
For example a packages core/auth has the same dependencies for dev like many other packages:

```{
  "name": "@core/auth",
  "description": "Authentication",
  "version": "1.0.0",
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "module": "./dist/index.mjs",
  "files": [
    "dist/**"
  ],
  "license": "MIT",
  "scripts": {
    "build": "tsup src/index.tsx --format esm,cjs --dts --external react",
    "dev": "tsup src/index.tsx --format esm,cjs --dts --external react",
    "clean": "rm -rf dist",
    "typecheck": "tsc --project ./tsconfig.json --noEmit"
  },
  "dependencies": {
    "remix-auth": "latest",
    "remix-auth-socials": "latest",
    "remix-auth-twitter": "latest"
  },
  "devDependencies": {
    "@core/tsconfig": "*",
    "@types/node": "^18.16.0",
    "@types/react": "^18.0.28",
    "@types/react-dom": "^18.0.11",
    "concurrently": "^7.6.0",
    "@core/eslint-config-vite": "*",
    "react": "^18.2.0",
    "rimraf": "^5.0.0",
    "tsup": "^6.7.0",
    "tsx": "^3.12.6",
    "typescript": "^5.0.4"
  }
}
``` 

Is there a command to update all packages across all packages?

**Comment by Jarrodsz** [comment]
To add i have like 5/6 packages and there is a lot of duplicate dev dependencies for each of them defining the same devDependencies block in all of them is there a way to have this more dry since it seems quite logical if one needs to update packages manually one forget or one gets version number discrepancies.

**Comment by PhilDL** [comment]
### Standard upgrade with `pnpm`

Hello, I usually do an upgrade with the recursive flag from the **root** of my monorepo

```
pnpm upgrade --interactive --latest --recursive
```

Then you get to select which packages to upgrade, for example:

```
Scope: all 6 workspace projects
? Choose which packages to update (Press <space> to select, <a> to toggle all, <i> to invert selection) …
  ● @types/node (dev)                      18.15.13 ❯ 18.16.3
  ● @typescript-eslint/eslint-plugin (dev)   5.59.0 ❯ 5.59.2
  ● @typescript-eslint/parser (dev)          5.59.0 ❯ 5.59.2
❯ ○ jose                                     4.14.1 ❯ 4.14.4
  ● prettier (dev)                            2.8.7 ❯ 2.8.8
  ● vite (dev)                                4.3.1 ❯ 4.3.4
  ● yargs                                    17.7.1 ❯ 17.7.2

Enter to start updating. Ctrl-c to cancel.
```
Here `@typescript-eslint/parser` is a dev dependencies in 6 packages and will be upgraded everywhere.

### With `manypkg`

https://github.com/Thinkmill/manypkg is an awesome tool that you can install at the root with `-w` flag and that will check and fixes mismatching package versions in underlying packages.

**Reply by Jarrodsz** [reply]
I thought it must be simple but could not figure it out searched the wrong keywords, thank you very much Phil and also for this repro it has been a tremendous help so far.
